### PR TITLE
gomlib: use deque instead of list for walk()

### PR DESCRIPTION
**Pull request by AstraLuma**
Exactly what it says on the tin.

This should be a performance improvement.

### CODE REVIEW

File: ppb/gomlib.py [hunk 1]
```diff
@@ -208,9 +208,9 @@ def walk(root):
 
     Is non-recursive.
     """
-    q = [root]
+    q = deque(root)
     while q:
-        cur = q.pop(0)
+        cur = q.pop()
```
Comment (pathunstrom) [inline ppb/gomlib.py]: Pretty sure this should be `pop_left`?

**Review by pathunstrom — APPROVED** [review]
bors r+

**Comment by AstraLuma** [comment]
bors retry

File: ppb/gomlib.py [hunk 2]
```diff
@@ -208,9 +208,9 @@ def walk(root):
 
     Is non-recursive.
     """
-    q = [root]
+    q = deque(root)
```
Comment (pathunstrom) [inline ppb/gomlib.py]:
```suggestion
    q = deque([root])
```

**Pull request merged**